src/java.desktop/share/classes/sun/awt/SunToolkit.java

```diff
@@ -1566,6 +1557,9 @@ private boolean isEQEmpty() {
      */
     @SuppressWarnings("serial")
     private final boolean waitForIdle(final long end) {
+        if (timeout(end) <= 0) {
+            return false;
+        }
```

Comment:
If the timeout is ended we will exit at the end of this method immediately, but let's post one less event to the EDT.